exist-core/src/main/java/org/exist/xquery/util/SerializerUtils.java

```diff
@@ -283,43 +397,43 @@ private static boolean checkTypes(final ParameterConvention parameterConvention,
         return false;
     }
 
-    private static void setPropertyForMap(final Properties properties, final ParameterConvention parameterConvention, final Sequence parameterValue) throws XPathException {
-        if(Type.BOOLEAN == parameterConvention.type) {
+    private static void setPropertyForMap(final Properties properties, final ParameterConvention<?> parameterConvention, final Sequence parameterValue) throws XPathException {
+        if(Type.BOOLEAN == parameterConvention.getType()) {
             // ignore "admit" i.e. "standalone" empty sequence
             if(!parameterValue.isEmpty()) {
                 if (((BooleanValue) parameterValue.itemAt(0)).getValue()) {
-                    properties.setProperty(parameterConvention.parameterName, "yes");
+                    properties.setProperty(parameterConvention.getLocalParameterName(), "yes");
                 } else {
-                    properties.setProperty(parameterConvention.parameterName, "no");
+                    properties.setProperty(parameterConvention.getLocalParameterName(), "no");
                 }
             }
-        } else if(Type.STRING == parameterConvention.type) {
+        } else if(Type.STRING == parameterConvention.getType()) {
             // ignore "absent" i.e. empty sequence
             if(!parameterValue.isEmpty()) {
-                properties.setProperty(parameterConvention.parameterName, ((StringValue) parameterValue.itemAt(0)).getStringValue());
+                properties.setProperty(parameterConvention.getLocalParameterName(), ((StringValue) parameterValue.itemAt(0)).getStringValue());
             }
-        } else if(Type.DECIMAL == parameterConvention.type) {
-            properties.setProperty(parameterConvention.parameterName, ((DecimalValue) parameterValue.itemAt(0)).getStringValue());
-        } else if(Type.QNAME == parameterConvention.type) {
+        } else if(Type.DECIMAL == parameterConvention.getType()) {
```

Comment:
the remaining code-smell is between this line and ~430. I agree with the tool that this section is very hard to maintain and follow. There are two options either refactor the code to be less complex, or add some inline comments so that others can follow the logic more easily. What do you think?